packages/job-worker/src/blueprints/context/RundownActivationContext.ts

```diff
@@ -0,0 +1,78 @@
+import { IRundownActivationContext, TSR } from '@sofie-automation/blueprints-integration'
+import { PeripheralDeviceId } from '@sofie-automation/shared-lib/dist/core/model/Ids'
+import { PeripheralDevicePublicWithActions } from '@sofie-automation/shared-lib/dist/core/model/peripheralDevice'
+import { literal } from '@sofie-automation/shared-lib/dist/lib/lib'
+import { Complete } from '@sofie-automation/corelib/dist/lib'
+import { PeripheralDeviceType } from '@sofie-automation/shared-lib/dist/peripheralDevice/peripheralDeviceAPI'
+import { ReadonlyDeep } from 'type-fest'
+import { JobContext, ProcessedShowStyleCompound } from '../../jobs'
+import { executePeripheralDeviceAction } from '../../peripheralDevice'
+import { CacheForPlayout } from '../../playout/cache'
+import { RundownEventContext } from './RundownEventContext'
+import { DBRundown } from '@sofie-automation/corelib/dist/dataModel/Rundown'
+import { PeripheralDevice } from '@sofie-automation/corelib/dist/dataModel/PeripheralDevice'
+
+export class RundownActivationContext extends RundownEventContext implements IRundownActivationContext {
+	private readonly _cache: CacheForPlayout
+	private readonly _context: JobContext
+
+	constructor(
+		context: JobContext,
+		cache: CacheForPlayout,
+		showStyleCompound: ReadonlyDeep<ProcessedShowStyleCompound>,
+		rundown: ReadonlyDeep<DBRundown>
+	) {
+		super(
+			context.studio,
+			context.getStudioBlueprintConfig(),
+			showStyleCompound,
+			context.getShowStyleBlueprintConfig(showStyleCompound),
+			rundown
+		)
+
+		this._context = context
+		this._cache = cache
+	}
+
+	async listPeripheralDevices(): Promise<PeripheralDevicePublicWithActions[]> {
```

Comment:
This code looks duplicated from the other context? It would be good to refactor it to be shared